epyqlib/device.py

```diff
@@ -529,26 +537,32 @@ def _init_from_zip(
                     break
 
         # TODO error dialog if no .epc found in zip file
-        filename = None
+        filename_epc = None
+        filename_sha = None
         for directory, directories, files in os.walk(path):
             for f in files:
                 logger.info(f)
                 if os.path.splitext(f)[1] == ".epc":
-                    filename = os.path.join(path, directory, f)
-                    break
+                    filename_epc = os.path.join(path, directory, f)
```

Comment:
I think that the an embedded `.epc` name won't be very interesting.  If a `.epc` is directly selected then use its path.  If a `.epz` is selected then use its path.  If I remember correctly, the build-generated `.epz` files get extra info in their name but the `.epc` files archived inside do not.

Reply:
I don't understand your request. For this section, I renamed the `filename` variable to `filename_epc` in this code section only to differentiate it from the newly added `filename_sha` variable. The `filename_epc` variable is only used locally. The Information tab logic performs as you describe: if load `.epc`, then display .`epc` file full path; if load `.epz`, then display `.epz` file full path.
Let me know if I'm missing something.

Reply:
I tried it out and see it works.  Now I look back and the `filename_epc` just has nothing to do with what gets shown in the info box...  sorry for the distraction here.